code/go/0chain.net/blobbercore/handler/object_operation_handler.go

```diff
@@ -1081,3 +1087,24 @@ func getExistingFileRef(fsh *StorageHandler, ctx context.Context, r *http.Reques
 	}
 	return nil
 }
+
+// check upload path is a file
+// ex := /path1/path2/path3/name
+// checking if /path1/path2/path3 is a file or dir
+func isUploadPathAFile(ctx context.Context, r *http.Request, allocationObj *allocation.Allocation, fileOperation string) bool {
+	if fileOperation == constants.FileOperationInsert {
+		var formData allocation.UpdateFileChanger
+		uploadMetaString := r.FormValue(getFormFieldName(fileOperation))
+		err := json.Unmarshal([]byte(uploadMetaString), &formData)
+		if err == nil {
+			fileDir := filepath.Dir(formData.Path)
+			if fileDir != "/" && fileDir != "" {
+				dirReference, _ := reference.GetReference(ctx, allocationObj.ID, fileDir)
```

Comment:
it is not worth to get whole reference record. it is enough to  read  `type` only.